plugins/outputs/file/file.go

```diff
@@ -39,11 +68,42 @@ func (f *File) SetSerializer(serializer serializers.Serializer) {
 	f.serializer = serializer
 }
 
+func (f *File) Init() error {
+	if len(f.Files) == 0 {
+		f.Files = []string{"stdout"}
+	}
+
+	if f.CompressionAlgorithm == "" {
+		return nil
+	} else if f.CompressionLevel == 0 {
+		switch f.CompressionAlgorithm {
+		case "zstd":
+			f.CompressionLevel = 3
+		case "zlib":
+			f.CompressionLevel = -1
+		case "gzip":
+			f.CompressionLevel = -1
+		}
+	}
+	err := validateCompressionAlgorithm(f.CompressionAlgorithm)
+	if err != nil {
+		return err
+	}
+	err = validateCompressionLevel(f.CompressionAlgorithm, f.CompressionLevel)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func (f *File) Connect() error {
+	var err error
 	writers := []io.Writer{}
 
-	if len(f.Files) == 0 {
-		f.Files = []string{"stdout"}
+	f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm, internal.EncoderCompressionLevel(f.CompressionLevel))
+	if err != nil {
+		return err
 	}
```

Comment:
I think this can be done in `Init()`.

Reply:
I think it's better to keep it in Connect as I've to write it twice in Init.

Reply:
WDYT? @srebhan

Reply:
Why do you have to write it twice? Just init the encoder and done...

Reply:
I had an if statement that returns nil but I don't actually need it anymore I can remove it.

Reply:
Sorry I currently cannot follow. Please submit the code and I take a look...

Reply:
https://github.com/influxdata/telegraf/blob/09adec7395366698bd20225b29eb807fe042037b/plugins/outputs/file/file_test.go#L33

There's a test here that fails because the file isn't running Init() I'm not sure if it's intended or not.

Reply:
Init should be called in that test indeed.